src/credssp/sspi_cred_ssp/tls_connection.rs

```diff
@@ -318,17 +353,24 @@ impl TlsConnection {
             TlsConnection::Rustls(tls_connection) => {
                 let connection_cipher = tls_connection
                     .negotiated_cipher_suite()
-                    .ok_or_else(|| Error::new(ErrorKind::InternalError, "Connection cipher is not negotiated"))?;
+                    .ok_or_else(|| Error::new(ErrorKind::InternalError, "connection cipher is not negotiated"))?;
 
-                let bulk_cipher = match connection_cipher {
-                    rustls::SupportedCipherSuite::Tls12(cipher_suite) => &cipher_suite.common.bulk,
-                    rustls::SupportedCipherSuite::Tls13(cipher_suite) => &cipher_suite.common.bulk,
+                let suite = match connection_cipher {
+                    rustls::SupportedCipherSuite::Tls12(cipher_suite) => &cipher_suite.common.suite,
+                    rustls::SupportedCipherSuite::Tls13(cipher_suite) => &cipher_suite.common.suite,
                 };
-                let block_size = match bulk_cipher {
-                    rustls::BulkAlgorithm::Aes128Gcm => AES_BLOCK_SIZE,
-                    rustls::BulkAlgorithm::Aes256Gcm => AES_BLOCK_SIZE,
+
+                let block_size = match suite.as_str() {
+                    Some(name) if name.contains("AES_128_GCM") => AES_BLOCK_SIZE,
+                    Some(name) if name.contains("AES_256_GCM") => AES_BLOCK_SIZE,
                     // ChaCha20 is a stream cipher
-                    rustls::BulkAlgorithm::Chacha20Poly1305 => 0,
+                    Some(name) if name.contains("CHACHA20_POLY1305") => 0,
+                    _ => {
+                        return Err(Error::new(
+                            ErrorKind::UnsupportedFunction,
+                            format!("cipher suite {suite:?} not supported"),
+                        ))
+                    }
                 };
```

Comment:
@TheBestTvarynka rustls API significantly changed in a breaking way and the `BulkAlgorithm` enum does not exist anymore. For the purpose of making this code compile again, I used this dirty trick, but I don’t think it’s a very good idea. Do you think you could look for a proper solution here?

Reply:
> Do you think you could look for a proper solution here?

Yes, I can. Should I prioritize it over my current tasks or it can wait a little?

Reply:
Thank you!
If you're almost done, go ahead with your current tasks, otherwise it would be good to take care of that in priority because I can't cut a new release with this